collectors/apicmcpfault.py

```diff
@@ -0,0 +1,58 @@
+import logging, re
+
+from prometheus_client.core import CounterMetricFamily, Summary
+import BaseCollector
+
+LOG = logging.getLogger('apic_exporter.exporter')
+REQUEST_TIME = Summary('network_apic_mcp_fault_counter',
+                       'Time spent processing request')
+
+
+class ApicIPsCollector(BaseCollector.BaseCollector):
+    def describe(self):
+        yield CounterMetricFamily('network_apic_mcp_fault_counter',
+                                  'Counter for MCP Faults')
+
+    @REQUEST_TIME.time()
+    def collect(self):
+        LOG.debug('Collecting APIC MCP Fault metrics ...')
+
+        g_dip = CounterMetricFamily(
+            'network_apic_mcp_fault_counter',
+            'Counter for MCP Faults',
+            labels=['apicHost', 'ip', 'fault_lifecyle', 'fault_summary', 'fault_desc'])
+
+        metric_counter = 0
+        query = "/api/node/class/faultInst.json?query-target-filter=or(eq(faultInst.code,\"F2533\"),eq(faultInst.code,\"F2534\"))"
+        for host in self.hosts: 
+            fetched_data = self.connection.getRequest(host, query)
+            if not self.connection.isDataValid(fetched_data):
+                LOG.warning(
+                    "Skipping apic host %s, %s did not return anything", host,
+                    query)
+            continue
+
+            if len(fetched_data['imdata']) == 0:
+            output = json.loads(query.text)
+            # Add Empty Counter to have the metric show up in Prometheus.
+            # Otherwise they only show when something is wrong and we dont know if it is actually working
+            g_dip.add_metric(labels=[host, '', '', ''], value=0)
+            metric_counter += 1
+            break  # Each host produces the same metrics.
```

Comment:
Lines 36-41 need to be intended further so that they are part of the the `if` from line 35